### PR TITLE
ORC-1331: Improve `PyArrow` page

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to improve [PyArrow](https://orc.apache.org/docs/pyarrow.html) page by adding `compression` codec option and a direct link to [Apache Arrow ORC](https://arrow.apache.org/docs/python/orc.html).

### Why are the changes needed?

To improve Python ORC user experience.

### How was this patch tested?

Manual review.

### CODE REVIEW

File: site/_docs/pyarrow.md [hunk 1]
```diff
@@ -9,25 +9,25 @@ permalink: /docs/pyarrow.html
 [Apache Arrow](https://arrow.apache.org) project's [PyArrow](https://pypi.org/project/pyarrow/) is the recommended package.
 
 ```
-pip3 install pyarrow==7.0.0
+pip3 install pyarrow==10.0.1
```
Comment (dongjoon-hyun) [inline site/_docs/pyarrow.md:12]: This is the latest PyArrow version.

File: site/_docs/pyarrow.md [hunk 2]
```diff
@@ -9,25 +9,25 @@ permalink: /docs/pyarrow.html
...
 pip3 install pandas
 ```
 
 ## How to write and read an ORC file
 
 ```
-In [1]: import pandas as pd
```
Comment (dongjoon-hyun) [inline site/_docs/pyarrow.md:19]: We don't need to import `pandas` directly.

File: site/_docs/pyarrow.md [hunk 3]
```diff
@@ -9,25 +9,25 @@ permalink: /docs/pyarrow.html
...
-Out[5]:
+In [4]: orc.read_table("test.orc").to_pandas()
+Out[4]:
    col1
 0     1
 1     2
 2     3
 ```
+
+[Apache Arrow ORC](https://arrow.apache.org/docs/python/orc.html) page provides more information.
```
Comment (dongjoon-hyun) [inline site/_docs/pyarrow.md:33]: This will lead the users to the official page with the full information.

**Comment by dongjoon-hyun** [comment]
cc @williamhyun

**Review by wgtmac — APPROVED** [review]
LGTM. Thanks @dongjoon-hyun!

**Comment by dongjoon-hyun** [comment]
Thank you so much, @wgtmac !

**Pull request closed without merging**